config/config.go

```diff
@@ -168,51 +169,66 @@ func (cfg *Config) addTransports(h host.Host) error {
 		// Should probably skip this if no transports.
 		return fmt.Errorf("swarm does not support transports")
 	}
-	var secure sec.SecureMuxer
+
+	muxers := make([]protocol.ID, 0, len(cfg.Muxers))
```

Comment:
If I understand correctly, the idiomatic fx way to do this is to use [named values](https://pkg.go.dev/go.uber.org/fx#hdr-Named_Values). While applying a name annotation for the result here is easy, it's not for the input to the upgrader constructor.

Not really sure what to do here.

Reply:
Is the constructor to the upgrader public? We could use a struct param

Reply:
> We could use a struct param

I don't really want to do that. If our dependency injection framework requires us to do ugly stuff all over the code base, I feel like we'd be doing things wrong.

---

With #1854, we'll need to pass the muxer along with its ID to the upgrader. This will resolve the potential for confusion in the future here. I suggest merging this PR as is, given that the fix for that issue is imminent.

Reply:
true that makes sense to me 👍